docs/model/index.md

```diff
@@ -0,0 +1,36 @@
+# farmOS Data Model
+
+## Goals
+
+The goal of the farmOS data model is to represent and organize a wide variety
+of agricultural and ecological systems to allow for easy access and analysis.
+This ranges from small-scale garden records to industrial agriculture to
+national forestry management.
+
+farmOS can model data that was recorded via manual data entry, as well as data
+streams from sensors or other applications.
+
+## Data types
+
+Data is organized into high-level data types called entities. The two primary
```

Comment:
I believe I've raised this concern in past conversations, but I wonder if it's wise to adopt Drupal terminology into our own model for farmOS. I'm worried this tight coupling with the Drupal implementation could lead to confusion down the road.

Specifically, I'm worried about how the farmOS concept of "entity" could evolve in the future, particularly if it's implemented in other languages and frameworks to mean something different from what a Drupal "entity" entails. Similarly, though probably less likely, Drupal's concept of "entity" could change over time. This could come to a head if and when we have farmOS entities that are not implemented as Drupal entities, and/or Drupal entities in farmOS which are not strictly farmOS entities. This could get confusing real fast.

Two alternatives that briefly come to mind: farm elements, or farm objects.

Reply:
Thanks for pointing that out @jgaehring ! I agree.

For what it's worth, I intentionally named the section "Data types" to avoid using the word "entities", but in searching through it looks like I did use "entity" in a few places (old habits die hard). I'll push a follow-up commit that changes those.

> Two alternatives that briefly come to mind: farm elements, or farm objects.

How about "records"? I've used this as an alternative to entities in presentations I've given in the past. I agree "entity" is too Drupal-specific, and sounds more technical/complicated than it needs to. "Records" is nice also because it fits the domain model. "Record keeping"! :-)

Reply:
Similarly, there are three references to "bundles" that I can find, which are not really necessary. I'll remove those too.